bentoml/frameworks/transformers.py

```diff
@@ -0,0 +1,196 @@
+    >>> from bentoml.frameworks.transformers import TransformersModelArtifact
+    >>>
+    >>> # Example : text generation using GPT2
+    >>> # Explicitly add either torch or tensorflow dependency
+    >>> # which will be used by transformers
+    >>> @bentoml.env(pip_packages=["torch==1.6", "transformers==3.1"])
+    >>> @bentoml.artifacts([TransformersModelArtifact('gptModel')])
+    >>> class TransformersService(bentoml.BentoService):
+    >>>     @bentoml.api(input=JsonInput())
+    >>>     def predict(self, parsed_json):
+    >>>          src_text = parsed_json[0].get("text")
+    >>>          model = self.artifacts.gptModel.get("model")
+    >>>          tokenizer = self.artifacts.gptModel.get("tokenizer")
+    >>>          input_ids = tokenizer.encode(src_text, return_tensors="pt")
+    >>>          output = model.generate(input_ids, max_length=50)
+    >>>          output = [tokenizer.decode(output[0], skip_special_tokens=True)]
+    >>>          return output
+    >>>
+    >>>
+    >>> svc = TransformersService()
+    >>> ts = TransformersGPT2TextGenerator()
+    >>> tokenizer = AutoTokenizer.from_pretrained("gpt2")
+    >>> model = AutoModelWithLMHead.from_pretrained(
+    >>>          "gpt2",
+    >>>          pad_token_id=tokenizer.eos_token_id)
+    >>> ts = TransformersGPT2TextGenerator()
+    >>> ts.pack("gptModel", {"model": model, "tokenizer": tokenizer})
+    >>>
+    >>> # OR - Directly pack the model by providing its name
+    >>> ts = ts.pack('gptModel','gpt2')
+    >>>
+    >>> # Note that while packing using the name of the model,
+    >>> # ensure that the model can be loaded using
+    >>> # transformers.AutoModelWithLMHead (eg GPT, Bert, Roberta etc.)
+    >>> # If this is not the case (eg AutoModelForQuestionAnswering, BartModel etc)
+    >>> # then pack the model by passing a dictionary
+    >>> # with the model and tokenizer declared explicitly
+    >>>
+    >>> saved_path = ts.save()
+    """
+
+    def __init__(self, name):
+        super(TransformersModelArtifact, self).__init__(name)
+        self._model = None
+        self._tokenizer_type = None
+        self._model_type = "AutoModelWithLMHead"
+
+    def _file_path(self, base_path):
+        return os.path.join(base_path, self.name)
+
+    def _load_from_directory(self, path):
+        if self._model_type is None:
+            raise NotFound(
+                "Type of transformers model not found. "
+                "This should be present in a file called "
+                "'_model_type.txt' in the artifacts of the bundle."
+            )
+        if self._tokenizer_type is None:
+            raise NotFound(
+                "Type of transformers tokenizer not found. "
+                "This should be present in a file called 'tokenizer_type.txt' "
+                "in the artifacts of the bundle."
+            )
+        transformers_model = getattr(
+            import_module("transformers"), self._model_type
```

Comment:
what's the reason for using `import_module` here?  is it for avoiding error when transformers not installed?

I think once it is removed from the `bentoml.artifact.__init__.py`, this should no longer be a problem.

Reply:
import_module is used to determine the type of model that has to be loaded from the transformers library since it'll keep varying